tests/system/Validation/ValidationTest.php

```diff
@@ -956,4 +957,110 @@ public function testTranslatedLabelTagReplacement()
 	}
 
 	//--------------------------------------------------------------------
+
+	/**
+	 * internal method to simplify placeholder replacement test
+	 * REQUIRES THE RULES TO BE SET FOR THE FIELD "foo"
+	 *
+	 * @param array|null $data optional POST data, needs to contain the key $placeholderField to pass
+	 */
+	protected function testPlaceholderReplacement(string $placeholder = 'id', ?array $data = NULL) {
+		if($data === NULL){
+			$data = [
+				$placeholder => 'placeholder-value'
+			];
+		}
+
+		$validationRules = $this->getPrivateMethodInvoker($this->validation, 'fillPlaceholders')($this->validation->getRules(), $data);
+		$fieldRules 	 = $validationRules['foo']['rules'] ?? $validationRules['foo'];
+		if (is_string($fieldRules)) {
+			$fieldRules = $this->getPrivateMethodInvoker($this->validation, 'splitRules')($fieldRules);
+		}
+
+		//loop all rules for this field
+		foreach ($fieldRules as $rule){
+			//only string type rules are supported
+			if(is_string($rule)){
+				$this->assertStringNotContainsString('{'.$placeholder.'}', $rule);
+			}
+		}
+	}
+
+	/**
+	 * method to test the test
+	 * @see ValidationTest::testPlaceholderReplacement()
+	 */
+	public function testPlaceholderReplacementTestFails()
+	{
+		$this->validation->setRule('foo', 'foo-label', 'required|filter[{id}]');
+
+		try {
+			$this->testPlaceholderReplacement('id', []);
+
+			//since we expect an exception, running through means failure
+			$this->fail('Failed to fail for wrong placeholder replacement');
+		} catch (ExpectationFailedException $exception){
+			//we expect this exception
+		}
```

Comment:
I'm not sure what's happening here. If you expect an exception to happen, then should you use `$this->expectException`?